gandalf.js

```diff
@@ -1,41 +1,45 @@
-/**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.